src/lib/libast/tests/string/stresc.c

```diff
@@ -0,0 +1,24 @@
+#include "config_ast.h"  // IWYU pragma: keep
+
+#include <string.h>
+
+#include "ast.h"
+#include "terror.h"
+
+tmain() {
+    UNUSED(argc);
+    UNUSED(argv);
+
+    // Very simple test to check if escape sequences are expanded
+    // `stresc()` internally uses `chresc()` function to expand escape sequences
+    // and it has more thorough tests.
+    char input[] = "\\n\\r";
+    char expected_result[] = "\n\r";
+
+    if (stresc(input) <= 0 || strcmp(input, expected_result)) {
```

Comment:
I would do the unescape in a separate step then verify the return value and include it if it's wrong; e.g.,

```
int rv = stresc(input);
if (rv != 2) terror("...", rv, 2);
```